_episodes/08-development-setup.md

```diff
@@ -1,138 +1,394 @@
+For more details about development installation, see ESMValTool documentation on
+[install from source][install-from-source].
+
+> ## Checking the development installation
+>
+> We can use the command ``conda list`` to list installed packages in
+> the ``esmvaltool`` environment.
+> Use this command to check that ESMValTool is installed in a ``develop`` mode.
+>
+> **Tips**: see the
+> [documentation on conda list](https://docs.conda.io/projects/conda/en/latest/commands/list.html).
+>
+>> ## Solution
+>>
+>> Run:
+>>
+>> ~~~bash
+>> conda list esmvaltool
+>>~~~
+>>
+>> ~~~bash
+>> # Name                    Version                   Build  Channel
+>> esmvaltool                2.1.1                     dev_0    <develop>
+>>~~~
+>>
+>> As can be seen, it is mentioned ``<develop>`` under the ``Channel``.
+> {: .solution}
+{: .challenge}
+
+## Contribution
 
-## Software installation
+ESMValTool is an open-source project in ESMValGroup.
+We can contribute to its development by:
 
-Once all prerequisites are fulfilled, ESMValTool can be installed by running
-the following commands in the directory containing the ESMValTool source code:
+- a new or updated recipe script, see lesson on
+[Writing your own recipe]({{ page.root }}{% link _episodes/05-preprocessor.md %})
+- a new or updated diagnostics script, see lesson on
+[Writing your own diagnostic script]
+- a new or updated cmorizer script, see lesson on
+[CMORization: Using observational datasets]
+- helping  with reviewing process of pull requests, see ESMValTool documentation on
+[Review of pull requests](https://docs.esmvaltool.org/en/latest/community/review.html)
+
+The next sections will explore the ways we can achieve this.
+
+### Review process
+
+We first discuss our idea in an
+**[issue](https://github.com/ESMValGroup/ESMValTool/issues)** in ESMValTool repository.
+Then, we create a new ``branch`` locally and start developing new codes.
+Once our development is finished, we can initiate a ``pull request``.
+For a full description of the GitHub workflow, please see ESMValTool documentation on
+[GitHub Workflow](https://docs.esmvaltool.org/en/latest/community/repository.html#github-workflow).
+
+The pull request will be tested, discussed and merged. This is called "**review process**".
+The process will take some effort and time to learn.
+However, a few “tools” i.e. command lines gets you a long way,
+and we’ll cover those essentials in the next sections.
+
+### Check code quality
+
+We aim to adhere to best practices and coding standards. The good news is that there are
+several tools that we can use to check our code against those standards.
+
+As an example, ``Pre-commit`` is a tool that checks your code for any invalid syntax
+and formatting errors. It also fixes some of those errors.
+To explore other tools, have a look at ESMValTool documentation on
+[Code style](https://docs.esmvaltool.org/en/latest/community/introduction.html#code-style).
```

Comment:
I added the list of the tools as you suggested. However, I didn't remove the link to the documentation as there is some info in the documentation that is beyond the scope of this lesson. Also, I think neither documentation nor this tutorial should cover details about these tools. Because they are related to best practices and code refinery lessons. We might just point users to those resources. This is also in line with what you mentioned about duplicate information.